### PR TITLE
Manage containerd as a systemd unit

### PR DESCRIPTION
**Pull request by joeatwork**
Changes from #2211, rebased on docker 1.11.2

~~Hold on review until #2218 is merged~~

### CODE REVIEW

File: app-emulation/docker/files/early-docker.service [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+Requires=containerd.service early-docker.socket
 
 [Service]
 Environment=TMPDIR=/var/tmp
-MountFlags=slave
-ExecStart=/usr/lib/coreos/dockerd daemon --host=fd:// --bridge=none --iptables=false --ip-masq=false --exec-root=/var/run/early-docker --graph=/var/lib/early-docker --pidfile=/var/run/early-docker.pid $DOCKER_OPTS $DOCKER_CGROUPS
+
+# We need shared mounts on the file system so that containerd can
+# see our bundles.
+# MountFlags=slave
```
Comment (marineam) [inline app-emulation/docker/files/early-docker.service]: Don't leave old config/code commented out, just delete.

File: app-emulation/docker/files/early-docker.service [hunk 2]
```diff
@@ -1,13 +1,17 @@
 [Unit]
 Description=Early Docker Application Container Engine
 Documentation=http://docs.docker.com
-After=early-docker.socket
-Requires=early-docker.socket
+After=containerd.service early-docker.socket
+Requires=containerd.service early-docker.socket
```
Comment (marineam) [inline app-emulation/docker/files/early-docker.service]: You are using the same containderd for both docker and early-docker. I seem to recall @mischief discovering that this isn't possible. Is that not the case?

Comment (joeatwork) [inline app-emulation/docker/files/early-docker.service]: I think I was able to observe this working - @mischief when you're around, can you talk me through the issues you had? It's not impossible that I misread the results of my (very manual, ad-hoc) testing, or didn't test the right cases.

**Comment by joeatwork** [comment]
Discussed IRL:

- Run flanneld with early-docker, as originally implemented, as a kola test. This will allow you to observe how early-docker actually behaves - networking conflicts between docker and early-docker may have been what shut this effort down back in the day

- Maybe just roll early-docker back to where it is on master if the current approach has *any* problems - early docker is deprecated and we don't need to add cool features to it.

- IF you have socket conflicts running early-docker with a child docker-containerd process, move the socket in use by the real-docker containerd process. This could have other beneficial effects!

File: app-emulation/docker/files/early-docker.service (hunk 2 above)
Comment (joeatwork) [inline app-emulation/docker/files/early-docker.service]: Discussed, see notes below

**Comment by joeatwork** [comment]
Superceeded by #2247

**Pull request closed without merging**